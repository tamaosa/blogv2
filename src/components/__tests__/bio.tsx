import React from "react"
import { useStaticQuery } from "gatsby"
import { render } from "@testing-library/react"
import Bio from "../bio"

describe("Bio", () => {
  const name = "tamaosa"
  const summary = "summary"
  const twitter = "twitter"
  const github = "github"
  beforeAll(() => {
    ;(useStaticQuery as jest.Mock).mockReturnValue({
      site: {
        siteMetadata: {
          author: {
            name: name,
            summary: summary,
          },
          social: {
            twitter: twitter,
            github: github,
          },
        },
      },
    })
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  it("snapshot test", () => {
    const { asFragment } = render(<Bio />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("matching name", () => {
    const { container } = render(<Bio />)
    expect(container).toHaveTextContent(name)
  })

  it("matching summary", () => {
    const { container } = render(<Bio />)
    expect(container).toHaveTextContent(summary)
  })

  it("testing link to github", () => {
    const { container } = render(<Bio />)
    expect(container.innerHTML).toMatch(
      `<a href="https://github.com/${github}" target="_blank" rel="external noopener" aria-label="GitHubアカウント">`
    )
  })

  it("testing link to twitter", () => {
    const { container } = render(<Bio />)
    expect(container.innerHTML).toMatch(
      `<a href="https://twitter.com/${twitter}" target="_blank" rel="external noopener" aria-label="Twitterアカウント">`
    )
  })
})
