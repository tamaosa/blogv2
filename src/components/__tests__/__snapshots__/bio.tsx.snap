// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio snapshot test 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

.emotion-1 {
  margin: 0 0.75rem 0 0;
}

.emotion-2 {
  margin: 0;
}

.emotion-2 span {
  font-weight: bold;
}

.emotion-2 a {
  color: var(--fg);
}

.emotion-2 div {
  margin: 0;
  font-size: 0.917rem;
  line-height: 1.5rem;
}

.emotion-3 {
  display: inline-block;
  width: 1.125rem;
  margin-left: 0.375rem;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
    <div
      class="emotion-2"
    >
      <span>
        tamaosa
      </span>
      <a
        aria-label="GitHubアカウント"
        href="https://github.com/github"
        rel="external noopener"
        target="_blank"
      >
        <span
          class="emotion-3"
        >
          <i
            class="fa"
          />
        </span>
      </a>
      <a
        aria-label="Twitterアカウント"
        href="https://twitter.com/twitter"
        rel="external noopener"
        target="_blank"
      >
        <span
          class="emotion-3"
        >
          <i
            class="fa"
          />
        </span>
      </a>
      <div>
        summary
      </div>
    </div>
  </div>
</DocumentFragment>
`;
